text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/Context.java

```diff
@@ -37,19 +39,48 @@
 @ApiStatus.NonExtendable
 public interface Context {
   /**
-   * Returns original message as provided to the parser.
+   * Parses a MiniMessage using all the settings of this context, including placeholders.
    *
-   * @return ogMessage
+   * @param message the message to parse
+   * @return the parsed message
    * @since 4.10.0
    */
-  @NotNull String originalMessage();
+  @NotNull Component parse(final @NotNull String message);
 
   /**
-   * Parses a MiniMessage using all the settings of this context, including placeholders.
+   * Create a new parsing exception.
    *
-   * @param message the message to parse
-   * @return the parsed message
+   * @param message the detail message
+   * @param tags the tag parts which caused the error
+   * @return the new parsing exception
    * @since 4.10.0
    */
-  @NotNull Component parse(final @NotNull String message);
+  @NotNull ParsingException newError(
+    final @NotNull String message,
+    final @NotNull ArgumentQueue tags
+  );
+
+  /**
+   * Create a new parsing exception without reference to a specific location.
+   *
+   * @param message the detail message
```

Comment:
its not clear if this is the error message or the input minimessage, that should be clarified in the javadocs (for all error methods here)